node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/task/util/editor/TextBufferImpl.java

```diff
@@ -104,12 +96,9 @@ public void delete(Position start, Position end) {
     }
 
     private void deleteLines(int startIndex, int endIndex) {
-        int fromIndex = endIndex;
-        for (int toIndex = startIndex; fromIndex <= getMaxLineIndex(); ++toIndex, ++fromIndex) {
-            lines.set(toIndex, lines.get(fromIndex));
+        for (int i = startIndex; i < endIndex; ++i) {
+            lines.remove(startIndex);
```

Comment:
Doesn't this, again, run in `O(nm)`, though, for `m` deletions in a list of `n` elements? That was my main reason for suggesting an iterator. Perhaps I'm just silly :)

Reply:
That's true. Reverted to former implementation due to using ArrayList, see comment elsewhere.